source/opt/scalar_replacement_pass.h

```diff
@@ -37,11 +38,10 @@ class ScalarReplacementPass : public MemPass {
  public:
   ScalarReplacementPass(uint32_t limit = kDefaultLimit)
       : max_num_elements_(limit) {
-    name_[0] = '\0';
-    strcat(name_, "scalar-replacement=");
-    const size_t name_len = strlen(name_);
-    snprintf(&name_[name_len], sizeof(name_) - name_len, "%d",
-             max_num_elements_);
+    const auto num_to_write = snprintf(
+        name_, sizeof(name_), "scalar-replacement=%u", max_num_elements_);
+    assert(size_t(num_to_write) <= sizeof(name_));
```

Comment:
Now that I am paying attention to the null, I believe `num_to_write` will not count the null that is needed.  I think the assert should be 

```static_cast<size_t>(number_to_write) < sizeof(name_))```

 Now I am wondering why we are even using null terminated strings instead of std::string.

Reply:
Indeed!  Good catch.

Reply:
Hilariously, I accounted for that extra 1 in my patch to protobufs. In fact it was the *only* reason to make the patch.
https://github.com/protocolbuffers/protobuf/pull/10128